web/client/components/data/query/QueryToolbar.jsx

```diff
@@ -263,14 +263,15 @@ class QueryToolbar extends React.Component {
         this.props.actions.onChangeDrawingStatus('clean', '', "queryform", []);
         this.props.actions.onReset();
         if (this.props.advancedToolbar) {
+            const isLgendFilterExist = this.props.appliedFilter?.filters?.find(f=>f.id === 'interactiveLegend');
             let filterObj = {
                 featureTypeName: this.props.featureTypeName,
                 groupFields: [],
                 filterFields: [],
                 spatialField: {
                     attribute: this.props.spatialField && this.props.spatialField.attribute
                 },
-
+                filters: isLgendFilterExist ? [isLgendFilterExist] : [],
```

Comment:
I think if we reset the QueryForm all the filters should be removed, I don't know if it's possible to communicate to the user that other external filter are currently applied and they will be removed on reset (cc @tdipisa )
Anyway I would prefer to remove this custom logic at the moment and reset all the filter

Reply:
> I don't know if it's possible to communicate to the user that other external filter are currently applied and they will be removed on reset (cc @tdipisa )

I think this one should be handled somehow for consistency and a better UX since, as far as I understand, all filters lends in the same filter object. What about something like this?
![image](https://github.com/geosolutions-it/MapStore2/assets/1280027/6560f44c-a49c-41ab-ad92-941529a83868)

or this?

![image](https://github.com/geosolutions-it/MapStore2/assets/1280027/57aafc78-d464-4f87-9ee4-1903ea12078f)